### ISSUE TITLE
Exemplars: different panels use different overlay widgets

### ISSUE THREAD

**Issue by costela** [grafana/grafana#53832]
**What happened**:

When displaying exemplars we noticed the Heatmap panel (right) uses a different widget than the Time-Series panel (left):

![image](https://user-images.githubusercontent.com/94699/185145277-5383e3fd-8d87-422c-a556-3e09ba2a5df8.png)

(screenshot from grafana 9.1.0)

**What you expected to happen**:

Ideally all panel types would share an common exemplar overlay. As a bonus, this widget should put the field used for linking at the top (as opposed to the bottom, as seen in the screenshot)

**How to reproduce it (as minimally and precisely as possible)**:

- Add some prometheus datasource with exemplar-storage turned on
- Collect different metric type with exemplars (e.g. counter + histogram)
- Display above metrics with different panel types

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.1.0
- Data source type & version: prometheus (builtin)
- OS Grafana is installed on: linunx
- User OS & Browser: Linux+Firefox / OSX+Chrome
- Grafana plugins:
- Others:

**Comment by leeoniya** [comment]
yes, this bit is an area of work in progress. the plan is to migrate thr exrmplar marker rendering to canvas like the new heatmap (so that we dont create hundreds or thousands of svg dom elements). part of this work will be unifying the tooltip/popover and refining the exemplar marker style.

**Comment by costela** [comment]
@leeoniya thanks for the update! Did I miss some issue tracking this progress, or is it being tracked internally?

**Comment by leeoniya** [comment]
> Did I miss some issue tracking this progress, or is it being tracked internally?

no, you didn't miss it.

we weren't quite sure how many people were using Exemplars with the old heatmap panel because of how :exploding_head: they were implemented. we definitely thought we had more time before anyone noticed! :sweat_smile:

i guess let's use this issue to track it! can't make any near-term commitments because there's a bunch of other work ahead of it and this will require refactoring at minimum TimeSeries tooltip, and writing an additional renderer with a spatial index. almost certainly a deeper rabbit hole than just the above, too!

**Comment by costela** [comment]
FYI: this issue technically still exists (widgets are different), though it has at least been partially solved for us because linking got added to the heatmap widget as well.
